src/PastJobs/PastJobs.js

```diff
@@ -79,147 +77,237 @@ const Job = ({job: {title, jobType, startDateTime, endDateTime, location, reques
             let jobItems = {...data};
+            var jobArray = [];
+            Object.keys(jobItems).map((key) => {
+                if (jobItems[key].title != null) {
+                    jobArray.push(jobItems[key]);
+                }
+            })
             this.setState({
-            jobs: sortBy(jobItems, 'date'),
+                jobs: jobArray.sort(function compare(a, b) {
+                    var dateA = new Date(a.startDateTime);
+                    var dateB = new Date(b.startDateTime);
+                    return dateA - dateB;
+                }),
             });
         });
         
         db.ref('/users').orderByChild('username').on('value', querySnapShot => {
-          let data = querySnapShot.val() ? querySnapShot.val() : {};
-          let userItems = {...data};
-          this.setState({
-              allUsers: sortBy(userItems, 'username'),
-          });
-      }); 
+            let data = querySnapShot.val() ? querySnapShot.val() : {};
+            let userItems = {...data};
+            this.setState({
+                allUsers: sortBy(userItems, 'username'),
+            });
+        });
     }
 
     getActiveUser(userKeys) {
-      let value = this.context;
-      for (var i = 0; i < userKeys.length; i++) {
-        var curr = this.state.allUsers[userKeys[i]];
-        if (curr.username == value["username"]) {
-          activeUser.username = curr.username;
-          activeUser.trustedUsers = curr.trustedUsers;
+        let value = this.context;
+        for (var i = 0; i < userKeys.length; i++) {
+            var curr = this.state.allUsers[userKeys[i]];
+            if (curr.username == value["username"]) {
+                activeUser.username = curr.username;
+                activeUser.trustedUsers = curr.trustedUsers;
+            }
         }
-      }
     } 
 
-      render() {
-        let jobsKeys = Object.keys(this.state.jobs);
+    render() {
         this.getActiveUser(Object.keys(this.state.allUsers));
       
         return (
-          <SafeAreaView style={styles.safeContainer}>
-              <ScrollView style={styles.scrollView}>
-              <View style={styles.container}>
-              {jobsKeys.length > 0 ? (
-                jobsKeys.map(key => (
-                  <Job
-                    key={key}
-                    id={key}
-                    job={this.state.jobs[key]}
-                  />
-                ))
-              ) : (
-                    <Text>No previous jobs</Text>
-              )}
-            </View>
-              </ScrollView>
-          </SafeAreaView>
+            <SafeAreaView style={styles.safeContainer}>
+                <ScrollView style={styles.scrollView}>
+                    {this.state.jobs.length > 0 ? (
+                        this.state.jobs.map((currentJob) => {
+                            if (currentJob.title != null) {
+                                let startJSONdate = new Date(currentJob.startDateTime);
```

Comment:
Why move this logic out of the job and into the render method??

Reply:
See response to comment on PastPosts page